### ISSUE TITLE
source code to your F-Droid build

### ISSUE THREAD

**Issue by eighthave** [iodeOS/fdroid#1]
The source code to the custom F-Droid builds in this repo must be available to all users who can download the APK, following the terms of the GNU GPL license.

Where can I find the source code?

**Comment by vincentvidal** [comment]
Hi, 
Here it is, since we forked it: https://gitlab.com/iode/os/apps/fdroid
There are links to the source code from several places, but maybe we should add something in that repo effectively.
You'll find the few modifications we made on the top, we constantly rebase on the official version. These are only small adaptations for the iodéOS rom, you won't find anything that might be of interest for you (they are related to adding our own fdroid repo and forcing repo updates for iodéOS users that were already running our rom before we added the repo). We did not change the branding, as we consider that the insignificant changes we made do not deserve to distribute it under a different name. Please let us know if that is a problem for you.
 We would be happy to contribute upstream, but put all our efforts on parts of iodéOS that really need improvements...
I take this opportunity to thank you for your work on F-Droid!

**Comment by eighthave** [comment]
Thanks, and congrats on https://iode.tech/en/news/iodeos-is-open-source/ !  That should open up more avenues for collaboration.  _fdroidclient_ is of course free software, and you can customize it as you like.  The problem is that this is a custom build that is shipped with the F-Droid branding and Application ID. Looking at the changes, I don't think iodéOS really needs any of them.  For example, about the change to _app/src/main/res/values/default_repos.xml_, have you seen the _additional_repos.xml_ functionality for Android ROMs?

About changing settings, we've been working with Calyx on making those kinds of things tie into the SetupWizard.

**Comment by vincentvidal** [comment]
Yes nothing is strictly required, and we could have avoided to fork the app if everything had been thought from the start. But, when we created our own f-droid repo, we wanted it to automatically appear in existing user's sessions (mainly to benefit from the automatic updates of other iodé's embedded apps and in particular microG). We also wanted to avoid that the icon disappeared from their launcher, which is why we kept the original package name. All that are the reasons of these (a bit nasty) hacks. We try to convince non-tecchie people to use alternative solutions for their smartphone, so we absolutely must avoid app icons that suddenly disappear and the need to enter settings menus (here, to set iodé f-droid repo) that are already too complicated for them.
To avoid confusion, we will add a readme in this github repo to warn that some of these apps are not the original ones, but changing package names for instance would break things for our users, it is too late now unfortunately.
We will think a bit about a better solution like setting parameters during Setup Wizard and making things a bit more clean, and let you know if you we need advices (or, simply, if we find an interesting way to collaborate).

**Comment by PanderMusubi** [comment]
@vincentvidal hi, where can find that readme? Thanks.
